### PR TITLE
[Gateway API] Add ingestion layer UTs

### PR DESCRIPTION
**Pull request by aaha97**
Add UTs for ingestion layer coverage

### CODE REVIEW

**Comment by Dhivyaaj** [comment]
No JIRA Ids found for the PR. Jira id is mandatory to update fix version in jira.                                            Please update respective Jira id in PR title or commit message                                            if the PR is intented for default branches of repo.                                            For mandatory exemptions, comment trigger phrase 'skip jira-id-check' in PR.                                            For manual trigger, comment trigger phrase 'run jira-id-check' in PR.

**Comment by akshayhavile** [comment]
skip jira-id-check

**Review by pkoshtavmware — APPROVED** [review]
LGTM

**Comment by akshayhavile** [comment]
build ako

File: tests/gatewayapitests/ingestion/gateway_test.go [hunk 1]
```diff
@@ -152,15 +159,245 @@ func TestGatewayCUD(t *testing.T) {
...
+	}
+	akogatewayapitests.SetGatewayGatewayClass(&gateway, gwClassName)
+
+	//create
+	gw, err := akogatewayapitests.GatewayClient.GatewayV1beta1().Gateways("default").Create(context.TODO(), &gateway, metav1.CreateOptions{})
+	if err != nil {
+		t.Fatalf("Couldn't create, err: %+v", err)
+	}
+	t.Logf("Created %+v", gw.Name)
+	waitAndverify(t, "")
```
[the remark this answers is not in the document]
Comment (aaha97) [reply tests/gatewayapitests/ingestion/gateway_test.go:197]: using status for gateway?

[the remark this answers is not in the document]
Comment (akshayhavile) [reply tests/gatewayapitests/ingestion/gateway_test.go:197]: we can take this later

**Review by akshayhavile — APPROVED** [review]

**Pull request merged**